examples/state_estimation_2d.py

```diff
@@ -236,21 +306,60 @@ def cost_weights_model():
                 print_stuff=epoch % 10 == 0 and i == 0,
             )
 
-        solution_path = get_path_from_values(
-            objective.batch_size, theseus_inputs, path_length
-        )
+        if learning_method == "leo":
```

Comment:
Add comment as discussed about high level explanation of LEO and reference to the paper and any specific equations.

Reply:
Added comments.